src/core/authorization_request/verification/did.rs

```diff
@@ -64,11 +78,12 @@ pub async fn verify_with_resolver(
         }
     }
 
-    let jwk = resolve_key(&kid, resolver)
+    let jwk = resolver
+        .fetch_public_jwk(Some(&kid))
         .await
-        .context("unable to resolve verification method from 'kid' header")?;
+        .context("unable to fetch JWK from 'kid' header")?;
 
-    let _: Json = ssi::jwt::decode_verify(&request_jwt, &jwk)
+    let _: Json = ssi::claims::jwt::decode_verify(&request_jwt, &jwk)
```

Comment:
Given the upgrade to `0.8.1`, the `resolve_key` method is removed.

I changed the argument type to `VerificationMethodDIDResolver` since it provides implementations of `DIDResolver` and `JWKResolver`.

That said, this is not working at the moment, resulting in the following error: 

```
thread 'w3c_vc_did_client_direct_post' panicked at tests/e2e.rs:49:58:
called `Result::unwrap()` on an `Err` value: unable to validate authorization request

Caused by:
    0: unable to validate Authorization Request
    1: unable to fetch JWK from 'kid' header
    2: internal error: DID resolution failed: DID method `key` not supported
```

My sense is that `fetch_public_key` isn't the right method to resolve the JWK.

Should I use a DIDResolver `Output` and parse the public key to create an `JWK` instance instead?

Reply:
I think that is just because you are using DID-JWK instead of DID-Key in the test:
![image](https://github.com/user-attachments/assets/234a849b-d61f-4474-bbc3-e8648d2f99bb)